src/wrapper.rs

```diff
@@ -0,0 +1,102 @@
+/* SPDX-License-Identifier: MIT */
+/*
+ * Copyright (C) 2023 IBM
+ *
+ * Authors:
+ *   Claudio Carvalho <cclaudio@ibm.com>
+ *   Vikram Narayanan <vikram186@gmail.com>
+ */
+
+#![allow(non_camel_case_types)]
+
+use crate::bindings::strlen;
+use crate::mem::sizeof_alloc;
+use crate::prints;
+use crate::util::util::memset;
+use alloc::alloc::{alloc, dealloc};
+use core::alloc::Layout;
+use core::slice;
+use core::{ptr, str};
+
+pub type c_char = u8;
+
+const ALIGN_32: usize = 32;
+
+#[no_mangle]
+pub extern "C" fn malloc(sz: usize) -> *mut u8 {
+    let layout = match Layout::from_size_align(sz, ALIGN_32) {
```

Comment:
I don't think you need to require a 32-byte alignment and you should use mem_allocate() here, e.g.:

  let va: VirtAddr = match mem_allocate(size) {
      Ok(v) => v,
      Err(()) => VirtAddr::zero(),
  };

  let ptr: *mut u8 = va.as_u64() as *mut u8;

  ptr